presto-main/src/main/java/com/facebook/presto/sql/relational/RowExpressionDomainTranslator.java

```diff
@@ -0,0 +1,903 @@
+import static com.facebook.presto.metadata.CastType.SATURATED_FLOOR_CAST;
+import static com.facebook.presto.spi.function.OperatorType.BETWEEN;
+import static com.facebook.presto.spi.function.OperatorType.EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.IS_DISTINCT_FROM;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.NOT_EQUAL;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.AND;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.IN;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.IS_NULL;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.OR;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static com.facebook.presto.sql.planner.LiteralEncoder.toRowExpression;
+import static com.facebook.presto.sql.relational.Expressions.call;
+import static com.facebook.presto.sql.relational.LogicalRowExpressions.FALSE;
+import static com.facebook.presto.sql.relational.LogicalRowExpressions.TRUE;
+import static com.facebook.presto.sql.relational.LogicalRowExpressions.and;
+import static com.facebook.presto.sql.relational.LogicalRowExpressions.or;
+import static com.facebook.presto.sql.relational.StandardFunctionResolution.getComparisonOperator;
+import static com.facebook.presto.sql.relational.StandardFunctionResolution.isComparisonFunction;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.collect.Iterators.peekingIterator;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+public final class RowExpressionDomainTranslator
+{
+    private final FunctionManager functionManager;
+    private final LogicalRowExpressions logicalRowExpressions;
+    private final StandardFunctionResolution functionResolution;
+
+    public RowExpressionDomainTranslator(FunctionManager functionManager)
+    {
+        this.functionManager = functionManager;
+        this.logicalRowExpressions = new LogicalRowExpressions(functionManager);
+        this.functionResolution = new StandardFunctionResolution(functionManager);
+    }
+
+    public RowExpression toPredicate(TupleDomain<VariableReferenceExpression> tupleDomain)
```

Comment:
Why this has to be specific for VariableReference? Can it be generic for all RowExpression?

Reply:
It has to be symbol